pkg/syncer/metadatasyncer.go

```diff
@@ -1719,6 +1721,73 @@ func pvcDeleted(obj interface{}, metadataSyncer *metadataSyncInformer) {
 	}
 }
 
+func pvAdded(obj interface{}, metadataSyncer *metadataSyncInformer) {
+	if metadataSyncer.clusterFlavor == cnstypes.CnsClusterFlavorWorkload && isPodVMOnStretchSupervisorFSSEnabled {
+		ctx, log := logger.GetNewContextWithLogger()
+		// Add owner reference of PV to its corresponding CNSVolumeInfo instance, if it doesn't exist.
+		pv, ok := obj.(*v1.PersistentVolume)
+		if pv == nil || !ok {
+			log.Warnf("pvAdded: unrecognized object %+v", obj)
+			return
+		}
+		log.Debugf("pvAdded: PV object %+v", pv)
+		if pv.Spec.CSI == nil {
+			log.Infof("PV %q is not a vSphere CSI volume. Skipping pvAdded functionality.", pv.Name)
+			return
+		}
+		if pv.Status.Phase != v1.VolumeBound {
```

Comment:
I think when PV is added it will not be directly in the bound state. It will be in the available state and then gets bound to the claim.

Reply:
Will it be the same case for static volume provisioning? If we are sure of this, I can remove the add event handler

Reply:
can you print the PV phase status in the PV Added event and confirm?